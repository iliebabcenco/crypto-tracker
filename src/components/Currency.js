import React from 'react';

const Currency = () => (
  <div>
    <h3>Currency</h3>
  </div>
);

export default Currency;
