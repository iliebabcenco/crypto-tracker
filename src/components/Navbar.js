import SearchForm from '../containers/SearchForm';
import '../styles/Navbar.css';

const Navbar = () => (
  <nav className="navbar">
    <h1 className="nav-title">CryptoTracker</h1>
    <SearchForm />
  </nav>
);

export default Navbar;
