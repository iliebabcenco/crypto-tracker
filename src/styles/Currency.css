.cur-logo {
  width: 3rem;
  max-height: 3rem;
  margin-right: 0.7rem;
}

.currency {
  border-bottom: 1px solid #c9ab81;
  width: 100%;
  margin-top: 1.5rem;
  padding-bottom: 1.5rem;
  font-size: 1.7rem;
  text-align: center;
  position: relative;
}

.currency-rank {
  text-align: left;
  position: absolute;
  left: 2%;
  font-size: 3rem;
}