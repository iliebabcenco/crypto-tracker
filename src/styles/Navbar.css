.navbar {
  width: 100%;
  color: white;
  display: flex;
  margin: 0 auto;
  padding-bottom: 1rem;
  padding-top: 1rem;
  justify-content: space-around;
  border-bottom: 1px solid #c9ab81;
  background-color: rgba(21, 30, 36, 0.8);
}