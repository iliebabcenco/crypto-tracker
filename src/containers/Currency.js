import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import { useDispatch } from 'react-redux';
import '../styles/Currency.css';

const Currency = ({ currency }) => {
  const dispatch = useDispatch();
  const handleClickGotoCurrency = (currency) => dispatch({ type: 'SHOW_CURRENCY', currency });
  return (
    <div className="currency">
      <span className="currency-rank">{currency.rank}</span>
      <img src={currency.logo_url} alt="logo" className="cur-logo" />
      <Link
        to={{
          pathname: '/currency',
          state: {
            currency,
          },
        }}
        key={currency.name}
        onClick={() => handleClickGotoCurrency(currency)}
      >
        {currency.name}
        {' '}
        {currency.symbol}

      </Link>
    </div>
  );
};

Currency.propTypes = {
  currency: PropTypes.shape({
    name: PropTypes.string.isRequired,
    logo_url: PropTypes.string.isRequired,
    symbol: PropTypes.string.isRequired,
    rank: PropTypes.string.isRequired,
  }).isRequired,
};

export default Currency;
