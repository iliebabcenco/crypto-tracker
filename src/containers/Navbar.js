import SearchForm from '../components/SearchForm';
import '../styles/Navbar.css';

const Navbar = () => (
  <nav className="navbar">
    <h1 className="nav-title">CryptoTracker</h1>
    <SearchForm />
  </nav>
);

export default Navbar;
