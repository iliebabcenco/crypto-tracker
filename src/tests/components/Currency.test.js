import React from 'react';
import { render, cleanup, screen } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import Currency from '../../components/Currency';
import store from '../../store/store';
import Navbar from '../../containers/Navbar';
import '@testing-library/jest-dom';

afterEach(cleanup);

const currency = {
  id: 'BTC',
  currency: 'BTC',
  symbol: 'BTC',
  name: 'Bitcoin',
  logo_url: 'https://s3.us-east-2.amazonaws.com/nomics-api/static/images/currencies/btc.svg',
  status: 'active',
  price: '25266.19061028',
  price_date: '2021-07-20T00:00:00Z',
  price_timestamp: '2021-07-20T08:42:00Z',
  circulating_supply: '18761506',
  max_supply: '21000000',
  market_cap: '474031786732',
  market_cap_dominance: '0.4262',
  num_exchanges: '385',
  num_pairs: '61017',
  num_pairs_unmapped: '5205',
  first_candle: '2011-08-18T00:00:00Z',
  first_trade: '2011-08-18T00:00:00Z',
  first_order_book: '2017-01-06T00:00:00Z',
  rank: '1',
  rank_delta: '0',
  high: '53388.87656362',
  high_timestamp: '2021-04-13T00:00:00Z',
  '1d': {
    volume: '22685513999.97',
    price_change: '-1457.58575551',
    price_change_pct: '-0.0545',
    volume_change: '6494930504.57',
    volume_change_pct: '0.4012',
    market_cap_change: '-27319940463.82',
    market_cap_change_pct: '-0.0545',
  },
  '30d': {
    volume: '903044940789.80',
    price_change: '-4699.05169091',
    price_change_pct: '-0.1568',
    volume_change: '-410992150281.06',
    volume_change_pct: '-0.3128',
    market_cap_change: '-87490634405.37',
    market_cap_change_pct: '-0.1558',
  },
};

it('should take a snapshot', () => {
  const { asFragment } = render(
    <Provider store={store}>
      <Router>
        <Currency currency={currency} />
      </Router>
    </Provider>,
  );
  expect(asFragment(
    <Provider store={store}>
      <Router>
        <Currency currency={currency} />
      </Router>
    </Provider>,
  )).toMatchSnapshot();
});

it('displays correct first child', () => {
  const { container } = render(
    <Provider store={store}>
      <Router>
        <Currency currency={currency} />
      </Router>
    </Provider>,
  );
  expect(container.firstChild).toMatchSnapshot();
});

it('negative check', () => {
  render(<Provider store={store}><Navbar /></Provider>);
  expect(screen.queryByText('Letâ€™s do some math!')).not.toBeInTheDocument();
});
