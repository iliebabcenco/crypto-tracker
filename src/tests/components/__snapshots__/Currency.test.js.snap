// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`displays correct first child 1`] = `
<div
  class="currency"
>
  <span
    class="currency-rank"
  >
    1
  </span>
  <img
    alt="logo"
    class="cur-logo"
    src="https://s3.us-east-2.amazonaws.com/nomics-api/static/images/currencies/btc.svg"
  />
  <a
    href="/currency"
  >
    Bitcoin
     
    BTC
  </a>
</div>
`;

exports[`should take a snapshot 1`] = `
<DocumentFragment>
  <div
    class="currency"
  >
    <span
      class="currency-rank"
    >
      1
    </span>
    <img
      alt="logo"
      class="cur-logo"
      src="https://s3.us-east-2.amazonaws.com/nomics-api/static/images/currencies/btc.svg"
    />
    <a
      href="/currency"
    >
      Bitcoin BTC
    </a>
  </div>
</DocumentFragment>
`;
